.github/workflows/ci_pull_request_open.yml

```diff
@@ -0,0 +1,56 @@
+name: CI Pull Request Open
+on:
+  pull_request:
+jobs:
+  ci-build-test-pr:
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: "-Xmx1500m -XX:+HeapDumpOnOutOfMemoryError"
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Maven
+        uses: stCarolas/setup-maven@v5
+        with:
+          maven-version: 3.9.6
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: oracle
+          cache: maven
+      - run: mvn --version
+#      - name: Build with Maven
+#        run: >
+#            mvn -s maven-settings.xml
+#            clean install surefire-report:report
+#            pmd:cpd pmd:pmd spotbugs:spotbugs
+#            -B -e -DskipTests -Pnative
+      - run: >
+            mvn -s maven-settings.xml clean install -DskipTests
+      - name: Test with Maven
+        run: >
+            mvn -s maven-settings.xml verify -B -e -DrerunFailingTestsCount=2
+            -Dmaven.test.failure.ignore=true -Dansi.strip=true
+      - name: Publish Test Report
+        if: success() || failure()
+        uses: ScaCap/action-surefire-report@v1
+      - name: Install xmlstarlet tool
+        if: success() || failure()
+        shell: bash
+        run: |
+            sudo apt-get update
+            sudo apt-get install -y xmlstarlet
+      - name: Publish Flaky Tests
+        if: success() || failure()
+        env:
+          TOKEN: ${{ secrets.JIRA_API_TOKEN }}
```

Comment:
Argh, this is a reason why we probably need to move to the bot model for publishing tests. It's not possible for secrets to be accessed when using `on.pull_request` as this potentially allows contributors to steal the secret value. For the existing automations we use `on.pull_request_target` which only checks out the source of the base branch, not the PR branch, and so prevents contributors performing such attacks. We can't use `on.pull_request_target` for CI builds, as obviously we need to checkout the source of the PR branch in order to test the changes.

Reply:
mmm yeah, it worked on my forked repo, but probably because pr is from the same repo